flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputProcessor.java

```diff
@@ -67,7 +63,8 @@ public StreamTwoInputProcessor(
 
 	@Override
 	public CompletableFuture<?> getAvailableFuture() {
-		if (inputSelectionHandler.areAllInputsSelected()) {
+		if (inputSelectionHandler.areAllInputsSelected() ||
+				(inputSelectionHandler.isInputUnavailable(0) && !inputSelectionHandler.isInputUnavailable(1))) {
```

Comment:
I don't get this condition. Why first has to be unavailable, while second available?

Reply:
Hah. That's actually a mistake. I forgot to remove the negation after I negated the method itself...